### DISCUSSION TITLE
[Feature] Customizable session token lifetime

### DISCUSSION THREAD

**Discussion by J3is** [immich-app/immich#13318] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

First of all, I'd like to thank all the developers for their work on this app. Immich is coming along great and I'm glad that people are able to use software like this free of charge.
I hope my wording of this doesn't make it seem like I'm requesting multiple features. Everything I list here are alternatives to get to the point where one can control how long sessions are, especially to avoid issues with OIDC providers.

#### The problem
From what I can see, the session cookie for Immich is hardcoded to have a lifetime of 400 days.
In my opinion, this is an absurdly large amount of time, and it can be a security problem in certain cases, for example when a user logs in from a public device and forgets to log out.

#### The proposed solution
Immich should provide such a long living token only if a "remember me" box is selected, and should honor token lifetime values when logging in via OIDC. Ideally, there should be an environment variable or a setting to configure the default session lifetime.

#### More details
When logging in via OIDC, Immich completely ignores the information it receives from the IdP about the duration of a session, which can create a lot of issues, since Immich also does not support either frontchannel or backchannel logout.
As an example, I have a custom page set as frontchannel logout URL in Keycloak, hosted under Immich's domain, which triggers a POST request to `/api/auth/logout` when opened. This request gets triggered only when logging out manually and not when Keycloak's session expires, since it expects the session lifetime values to be respected. This means that Immich remains logged on, and that the next time I log in to Keycloak, frontchannel logout won't be triggered even if I log out manually, since Keycloak is not aware anymore of the existence of Immich's session.

#### Alternatives
An alternative to this I believe would be to support something like backchannel logout, so an OIDC provider can disconnect the user when its session expires (I'm not sure though if backchannel logout would be triggered if the IdP thinks the session has already expired, but that could be configured on the IdP side I believe by changing the OIDC token lifetime). Making the session lifetime customizable would probably be much easier to implement though.

Thanks for reading! :)

### Platform

- [ ] Server
- [X] Web
- [ ] Mobile

**Comment by FitzroyZ** [comment]
same. sometimes I just logged on a family-shared device and forgot to log out. (just do some administration for scanning external lib)
really wish there is a 'remember me' option,
that i can close the browser and knowing everything is fine.

**Comment by FitzroyZ** [comment]
I use Caddy to reverse proxy the webserver. I found that you could config the session lifetime through the proxy.

example, for 14 days. 1209600 secs.

example.com {
    reverse_proxy http://immich-server:2283 {
          header_down Set-Cookie "(.+); Max-Age=\d+(.*)" "$1; Max-Age=1209600$2"
   }
    tls /certs/sample.crt /certs/sample.key
}
<img width="962" height="143" alt="Snipaste_2026-02-02_18-34-20" src="https://github.com/user-attachments/assets/16c9129c-5dc7-4a39-8e1c-501a7d4e8629" />

**Comment by bo0tzz** [comment]
#26235
